qos/qospb/qos.proto

```diff
@@ -0,0 +1,27 @@
+syntax = "proto3";
+package qospb;
+
+import "gogoproto/gogo.proto";
+
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.sizer_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+option (gogoproto.goproto_getters_all) = false;
+option (gogoproto.goproto_enum_prefix_all) = false;
+
+
+message Subject {
+  string name = 1;
+  string prefix = 2;
+}
+
+message QoSRule {
+  string rule_name = 1;
+  string rule_type = 2;
```

Comment:
I think there will be many types of rules eventually with different sets of settings they support. 
How about: 
```
 message QoSRule {
  string rule_name = 1;
  uint32 priority = 2 ;
  oneof rule_type {
      message GRpcTrafficRule grpcTrafficRule = 4;
      message CustomRule customRule = 5;
      ... 
  }
}

message gRpcTrafficRule {
   string rpcName = 1;
   string keyPrefix = 2;
   uint64 qps = 3;
   boolean per_user = 4;
} 
...
message CustomRule {
  string condition = 1;
}
```

This would self-document which rule-types support which settings and keep it expandable.

Reply:
good advice, thanks.i will optimize it.